DEVELOPMENT.md

```diff
@@ -0,0 +1,80 @@
+# Local Development Setup
+
+These instructions assume you have SDG cloned to a directory locally
+and are executing these commands within that directory.
+
+## Setup a Python virtual environment and install InstructLab
+
+```shell
+python3 -m venv --upgrade-deps venv
+. venv/bin/activate
+pip install git+https://github.com/instructlab/instructlab@main
```

Comment:
I worry it might overcomplicate the doc but if there's space, it might be worthwhile to recommend users to clone this locally too, that way they can edit both in parallel.

Reply:
We chatted about this in-person, but just recapping the conversation here. I'm fine with expanding this to account for more scenarios, but the decision to include instructions directly here vs linking to stable documentation in the instructlab/instructlab repo needs a bit of thought and to be done in a way that doesn't over-complicate the scenario of someone just wanting to work on SDG alone. I think that's a good thing to address in a follow-up PR if a strong opinion emerges.

Reply:
Let's keep it simple